plugins/esri_grid_ascii/main.py

```diff
@@ -115,93 +111,65 @@ def load_data(self):
             return self._current_grid.copy()
 
         path = os.path.abspath(self.path)
-        nodata_value = self._header['nodata_value']
-        if self._is_velma and self._temporal_info.is_temporal:
-
+        if self._is_velma and self.time_info.is_temporal:
             if date is None:
                 date = Timeline.app().current
 
-            start = self._temporal_info.timestamps[0]
-            end = self._temporal_info.timestamps[-1]
-
+            start = self.time_info.timestamps[0]
+            end = self.time_info.timestamps[-1]
             if date < start:
                 date = start
             elif date > end:
                 date = end
 
-            filename = "{}_{}".format(self._name, self._loop)
+            filename = "{}_{}".format(self.data_name, self._loop)
             if self._has_layer:
                 filename = filename + '_{}'.format(self._layer)
-
             filename = filename + "_{}_{}".format(date.year, date.timetuple().tm_yday)
             if self._is_subday:
                 filename = filename + '_{:2d}_{:2d}'.format(date.hour, date.minute)
             filename = "{}.asc".format(filename)
             path = os.path.join(os.path.dirname(path), filename)
 
-        data = numpy.loadtxt(path, skiprows=6).astype(numpy.float32)
-        data = numpy.ma.masked_where(data == nodata_value, data)    # Mask out nodata
-        self._current_grid = data
+        with rasterio.open(path) as src:
+            self._current_grid = src.read(1)
         self._current_variable = variable
         self._current_time = date
         return self._current_grid.copy()
 
     @property
-    def shape(self):
-        return self._header['nrows'], self._header['ncols']
-
-    @property
-    def resolution(self):
-        return self._header['cellsize']
-
-    @property
-    def extent(self):
-        return self._extent
-
-    @property
-    def time_info(self):
-        return self._temporal_info
-
-    @property
     def variables(self):
-        return [self._name]
+        return [self.data_name]
 
     def calculate_stats(self):
         if self.stats.is_stale:
             stats = VariableStats()
-            if self.time_info and self.time_info.is_temporal:
-                maxs = []
-                mins = []
-                for step in self.time_info.timestamps:
-                    grid = self.get_data("", step)
-                    mins.append(grid.min())
-                    maxs.append(grid.max())
-                stats.min_value = float(min(mins))
-                stats.max_value = float(max(maxs))
-            else:
-                grid = self.get_data("")
-                stats.min_value = float(grid.min())
-                stats.max_value = float(grid.max())
-
-            stats.nodata_value = self._header['nodata_value']
+            maxs = []
+            mins = []
+            steps = self.time_info.timestamps if self.time_info.is_temporal else [0]
+            for step in steps:
```

Comment:
Untested, but this may work:

```python
stats.min_value = min(float(self.get_data('', step).nanmin()) for step in steps)
# same for max
```

Reply:
Almost works, reason for the loop is because I need to filter out the `nodata_value` in the calculation, and it just makes a bit more sense to do it in the loop. Fun one-liner though if no masking is needed.

Reply:
That's what `nanmim` _should_ take care of...

Reply:
It should if the input has `numpy.nan`s in there, but the `rasterio` read isn't masked (currently):
```
    with rasterio.open(path) as src:
        self._current_grid = src.read(1)
```
It just returns the data in the clear without a mask. We could add a 'mask=True` parameter for Raster-based plugins to do this, since for plugins where we want to create a terrain out of the data, we don't want it masked since we handle that transformation ourselves.

Reply:
It seems like where appropriate, data should be masked. Using numpy, any code using the data that doesn't want to the mask for some reason or another can ignore it.